builtin/providers/aws/resource_aws_lambda_function_test.go

```diff
@@ -80,18 +90,21 @@ func testAccCheckAwsLambdaFunctionExists(n string, function *lambda.GetFunctionO
 	}
 }
 
-func testAccCheckAWSLambdaAttributes(function *lambda.GetFunctionOutput) resource.TestCheckFunc {
+func testAccCheckAWSLambdaAttributes(fname string, function *lambda.GetFunctionOutput) resource.TestCheckFunc {
 	return func(s *terraform.State) error {
 		c := function.Configuration
-		const expectedName = "example_lambda_name"
-		if *c.FunctionName != expectedName {
-			return fmt.Errorf("Expected function name %s, got %s", expectedName, *c.FunctionName)
+		if *c.FunctionName != fname {
+			return fmt.Errorf("Expected function name %s, got %s", fname, *c.FunctionName)
 		}
 
 		if *c.FunctionArn == "" {
 			return fmt.Errorf("Could not read Lambda Function's ARN")
 		}
 
+		if strings.HasSuffix(fname, "s3") && *c.Version != "$LATEST" {
+			return fmt.Errorf("Expected version $LATEST, got %s", *c.Version)
+		}
+
```

Comment:
It would be nice to separate this function out into more generic ones instead of hardcoding values, e.g.

 - `testAccCheckAWSLambdaFunctionName(expectedName string, function *lambda.GetFunctionOutput)`
 - `testAccCheckAWSLambdaFunctionVersion(expectedVersion string, function *lambda.GetFunctionOutput)`